### PR TITLE
Tron support

### PR DESCRIPTION
**Pull request by alexqrid**
This PR adds full Tron support:

- Tron main transactions, as well as dex transfers
- Tron internal transactions
- Tron TRC-10 transactions, as well as dex transfers
- Tron TRC-20 transactions
- Tron TRC-721 transactions
- Tron TRC-1155 transactions

Pay attention that main module works correctly only with the [following PR](https://github.com/tronprotocol/java-tron/pull/5469), as it fall backs to geth-like `JSON-RPC` for transaction types with undefined (in module code) processing rules, and as official node returns various currencies in `value` field of `JSON-RPC`, that PR must be applied to the node, that is used for the module.

I've also added some debug features: processing range of blocks, processing from the latest block down to genesis.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -655,6 +655,72 @@ MODULE_ton-main_NODES[]=http://login:password@127.0.0.2:1234/
 MODULE_ton-main_REQUESTER_TIMEOUT=60
 MODULE_ton-main_REQUESTER_THREADS=12
 
+#######################
+## Main TRON Module
+#######################
+
+MODULES[]=tron-main
+MODULE_tron-main_CLASS=TronMainModule
+MODULE_tron-main_NODES[]=http://login:password@127.0.0.1:1234/
```
Comment (alexqrid) [inline .env.example:664]: @Har01d please make sure that your nodes are behind proxy with the following rules:

-  `/` is proxied to node's `/jsonrpc` path
- `/wallet/` is proxied to node's `/wallet/` path

Comment (jzethar) [reply .env.example:664]: If it is necessary to use different passes to API, perhaps, it is better to use different environmental variables:
`MODULE_tron-main_JSONRPC[] = http://login:password@127.0.0.1:1234/jsonrpc/`
`MODULE_tron-main_WALLET[] = http://login:password@127.0.0.1:1234/wallet/`
And use in code:
```php
$tron_jsonrpc = envm($this->module, 'JSONRPC');
$info = requester_single($tron_jsonrpc,  params: ['jsonrpc' => '2.0', 'method' => 'eth_blockNumber', 'id' => 0], result_in: 'result', timeout: $this->timeout);
```

Comment (alexqrid) [reply .env.example:664]: Sounds reasonable, but @Har01d what do you think if I create additional class atribute in order to not duplicate `$tron_jsonrpc = envm($this->module, 'JSONRPC');` in every invocation of `jsonrpc` endpoint?

**Review by Har01d — APPROVED** [review]
TRON has been running in production for a while (https://3xpl.com/tron) and the database has been fully synced.

**Comment by alexqrid** [comment]
@Har01d Forgot to add last fix for querying TRC10 info on early blocks see https://github.com/3xplcom/Core/pull/57

**Pull request merged**